### PR TITLE
fix: project access permissions

### PR DESCRIPTION
**Pull request by ZeRego**
### Description:
- apply permission check to all ProjectService methods
- update permission checks in saved chart and dashboard services
- apply permission check in org appearance form
- apply permission check in all project setting forms
- some other small fixes around permissions

Screenshot 2022-06-22 at 16 41 20
Screenshot 2022-06-22 at 16 41 11
Screenshot 2022-06-22 at 16 40 56
Screenshot 2022-06-22 at 16 38 27
Screenshot 2022-06-22 at 16 30 44

### CODE REVIEW

File: packages/frontend/src/pages/Settings.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ const Settings: FC = () => {
                         )}
                         {orgData &&
                             !orgData.needsProject &&
-                            user.data?.ability?.can('manage', 'Project') && (
+                            user.data?.ability?.can('view', 'Project') && (
```
Comment (rephus) [inline packages/frontend/src/pages/Settings.tsx:108]: why this change ?

Comment (ZeRego) [reply packages/frontend/src/pages/Settings.tsx:108]: So member users can see the list of all the projects they have access to and access the project settings page. I instead have hidden the create and delete button if they don't have permissions to delete it.

**Review by rephus — APPROVED** [review]
